### PR TITLE
Cli option cwd

### PR DESCRIPTION
**Pull request by chrmod**
to be merged after: https://github.com/broccolijs/broccoli/pull/342

### CODE REVIEW

File: lib/cli.js [hunk 1]
```diff
@@ -63,38 +67,62 @@ module.exports = function broccoliCLI(args) {
...
+  return actionPromise || RSVP.resolve();
 };
 
 function getBuilder(options) {
   if (!options) {
     options = {};
   }
-  return new broccoli.Builder(broccoli.loadBrocfile(options.brocfilePath));
+  const brocfile = broccoli.loadBrocfile(options.brocfilePath);
+  process.chdir(options.cwd || brocfile.baseDir);
```
Comment (stefanpenner) [inline lib/cli.js]: I would have thought we would `chdir` before loading the brocfile, that way `cwd` is consistent before we touch user code.

Comment (chrmod) [reply lib/cli.js]: @stefanpenner was thinking about that and have mixed feelings. 

consider using `--brocfile-path` with `--cwd` together, example:
`broccoli serve --cwd <project-path> --brocfile-path ./Brocfile.development.js`
should the brocfile path be resolved to `<project-path>/Brocfile.development.js` or to `<current working dir>/Brocfile.development.js` ? 


another issue is `findup` in `load_brocfile.js` - will it be understandable that findup will be affected by `--cwd`?

Comment (stefanpenner) [reply lib/cli.js]: hmm, I'll have to think a bit more..

Comment (stefanpenner) [reply lib/cli.js]: I think `--cwd` should not affect the CLI arguments, but should affect things like `process.cwd()` in userland code. My mental model is that CLI parameters are evaluated in the context of the shell. This ensures the shells autocomplete works regardless of the `--cwd`. As if the user wants to change `cwd` in the shell, they should just do that...

So:

```sh
broccoli --cwd somewhere/else --brocfile-path ./Brocfile.development.js
```

```js
// Brocfile.js
process.cwd() // should be --cwd
```

Comment (chrmod) [reply lib/cli.js]: :+1: I do like the argument with shell autocompletion

**Comment by chrmod** [comment]
@stefanpenner I've updated the PR to change `cwd` behavior. Still have to update tests to make some sense. Please suggest if we should test something specific.

**Comment by chrmod** [comment]
@stefanpenner actually those tests kinda work. Please consider a merge

**Comment by rwjblue** [comment]
Thanks @chrmod !

**Pull request merged**